src/CoreVideo/CVDisplayLink.cs

```diff
@@ -57,6 +58,64 @@ internal CVDisplayLink (IntPtr handle, bool owns)
 			this.handle = handle;
 		}
 
+		[Mac (12,0)]
+		[DllImport (Constants.CoreVideoLibrary)]
+		static extern CVReturn CVDisplayLinkCreateWithCGDisplay (uint displayId, out IntPtr displayLink);
+
+		[Mac (12,0)]
+		public static CVDisplayLink? FromDisplayId (uint displayId, out CVReturn error)
+		{
+			error = CVDisplayLinkCreateWithCGDisplay (displayId, out IntPtr handle);
+			if (error != 0)
+				return null;
+
+			return new CVDisplayLink (handle, true);
+		}
+
+		[Mac (12,0)]
+		public static CVDisplayLink? FromDisplayId (uint displayId)
+			=> FromDisplayId (displayId, out var _);
+
+		[Mac (12,0)]
+		[DllImport (Constants.CoreVideoLibrary)]
+		unsafe static extern CVReturn CVDisplayLinkCreateWithCGDisplays (uint* displayArray, nint count, out IntPtr displayLink);
+
+		[Mac (12,0)]
+		public static CVDisplayLink? FromDisplayIds (uint[] displayIds, out CVReturn error)
+		{
+			if (displayIds == null)
+				throw new ArgumentNullException (nameof (displayIds));
+			error = 0;
+			IntPtr handle = IntPtr.Zero;
+			unsafe {
+				fixed (uint *displaysHandle = displayIds) {
+					error = CVDisplayLinkCreateWithCGDisplays (displaysHandle, displayIds.Length, out handle);
+				}
+			}
```

Comment:
If you change the P/Invoke signature above, this can be changed to:

```suggestion
			error = CVDisplayLinkCreateWithCGDisplays (displayIds, displayIds.Length, out handle);
```